src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1173,7 +1176,10 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
                     default:
                         break;
                 }
-                return conversions.HasAnyNullabilityImplicitConversion(sourceType, targetType);
+
+                Debug.Assert(conversions.IncludeNullability);
+                HashSet<DiagnosticInfo> discardedDiagnostics = null;
+                return conversions.ClassifyImplicitConversionFromType(sourceType, targetType, ref discardedDiagnostics).Kind != ConversionKind.NoConversion;
```

Comment:
:memo: This is the heart of this change. 
`HasAnyNullabilityImplicitConversion` checks `HasTopLevelNullabilityImplicitConversion` and `ClassifyImplicitConversionFromType`, but we no longer need to do the top-level check.
The top-level check is done by `NullableWalker.AreParameterAnnotationsCompatible`, which accounts for attributes in addition to annotations from types.